lib/Service/SystemTagService.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Assistant\Service;
+
+use OCP\SystemTag\ISystemTagManager;
+use OCP\SystemTag\ISystemTagObjectMapper;
+use OCP\SystemTag\TagNotFoundException;
+
+class SystemTagService {
+
+	public const AI_TAG_NAME = 'Generated using AI';
+
+	public function __construct(
+		private ISystemTagManager $systemTagManager,
+		private ISystemTagObjectMapper $systemTagObjectMapper,
+	) {
+	}
+
+	public function getAiTag() {
+		try {
+			return $this->systemTagManager->getTag(self::AI_TAG_NAME, true, true);
+		} catch (TagNotFoundException $e) {
+			return $this->systemTagManager->createTag(self::AI_TAG_NAME, true, true);
+		}
+	}
+
+	/**
+	 * @param string $fileId
+	 * @return void
+	 * @throws TagNotFoundException
+	 */
+	public function assignAiTagToFile(string $fileId) {
+		$this->systemTagObjectMapper->assignTags($fileId, 'files', $this->getAiTag()->getId());
```

Comment:
perhaps the tag can be created in a migration so we're sure it exists and then store the tag's ID in a config. It might be a little better to not create `ISystemTag` every time or is it not that impactful?

Reply:
> perhaps the tag can be created in a migration so we're sure it exists

can they be deleted by users? If yes, then ignore this review.